base/src/main/java/org/aya/tyck/ExprTycker.java

```diff
@@ -155,6 +157,45 @@ public final class ExprTycker extends Tycker {
         var items = tuple.items().map(this::synthesize);
         yield new TermResult(new IntroTerm.Tuple(items.map(Result::wellTyped)), new FormTerm.Sigma(items.map(item -> new Term.Param(Constants.anonymous(), item.type(), true))));
       }
+      case Expr.CoeExpr coe -> {
+        assert coe.resolvedVar() instanceof DefVar<?, ?> defVar
+          && defVar.core instanceof PrimDef def && def.id == PrimDef.ID.COE : "desugar bug";
+        var defVar = coe.resolvedVar();
+        var mockApp = new Expr.AppExpr(coe.sourcePos(), new Expr.AppExpr(coe.sourcePos(),
+          new Expr.RefExpr(coe.id().sourcePos(), defVar),
+          new Expr.NamedArg(true, coe.type())),
+          new Expr.NamedArg(true, coe.restr()));
+        var res = synthesize(mockApp);
+        if (whnf(res.wellTyped()) instanceof PrimTerm.Coe(var type, var restr)) {
+          var bad = new Object() {
+            Term typeSubst;
+            boolean stuck = false;
+          };
+          // in case the `type` was eta-expanded from a definition.
+          var typeNF = type.normalize(state, NormalizeMode.NF);
+          var freezes = CofThy.conv(restr, new Subst(), subst -> {
+            var typeSubst = typeNF.subst(subst);
+            // ^ `typeSubst` should now be instantiated under cofibration `restr`, and
+            // it must be the form of `(i : I) -> A`. We need to ensure the `i` does not occur in `A` at all.
+            // See also: https://github.com/ice1000/guest0x0/blob/main/base/src/main/java/org/aya/guest0x0/tyck/Elaborator.java#L293-L310
+            Function<Integer, Boolean> post = usages -> {
+              if (usages != 0) bad.typeSubst = typeSubst;
+              return usages == 0;
+            };
+            return switch (typeSubst) {
```

Comment:
Add a case for Error Term?

Reply:
I guess it is not needed because `CofThy.conv` does not callback when `restr` contains an error.

Reply:
I think we can skip the check when `type` is an error term because we cannot give any useful tips in this case.

Reply:
Sure